### ISSUE TITLE
Add Microsoft Edge

### ISSUE THREAD

**Issue by le717** [WordPress/browsehappy#24]
I'd love it if it would replace the IE listing, but that is sadly not possible (yet).

http://www.microsoft.com/en-us/windows/microsoft-edge

**Comment by chadmoller** [comment]
:thumbsup:

**Comment by coredumperror** [comment]
Another thumbs up from me. Though I don't think an Edge link should replace IE11; it should live alongside it. Not everyone has Windows 10, and you can't run Edge on 7 or 8.

**Comment by Otto42** [comment]
https://meta.trac.wordpress.org/ticket/1206

**Comment by mgol** [comment]
I see Edge now appears at http://browsehappy.com/. It shows as version 25, though. I think the main purpose of browsehappy is to advocate browsers with modern rendering engines and not the UI itself; in such cases Microsoft advises to use the EdgeHTML (i.e. the engine) version number instead of the app shell one; currently it's 13. This is the number used by, amongst others:
1. Can I Use: http://caniuse.com/#feat=template
1. HTML5 Test: https://html5test.com/results/desktop.html
1. Kangax ECMAScript compat tables: https://kangax.github.io/compat-table/es6/
1. The BrowserStack API: https://api.browserstack.com/4/browsers

Perhaps you should use this number as well?

More at https://blogs.windows.com/msedgedev/2015/09/21/understanding-versions-in-an-evergreen-browser/.

**Comment by coredumperror** [comment]
Agreed. I've never heard anything about Edge having any versions besides 12 and 13.

**Comment by coffee2code** [comment]
@mgol: While what you say is true, Browse Happy is intended for users and not developers (unlike the resources you cite). As such, they don't know (and may not be able to easily check, yet) the version of EdgeHTML being used.

But that doesn't really matter to them. The latest Edge would include the latest available version of EdgeHTML. An update to EdgeHTML that wasn't included with Edge (say EdgeHTML bumped to 14 while Edge was still on 27) is useless to report as it is non-actionable.

For UA inspection, standards compliance testing and reporting, and other developer-focused concerns, the EdgeHTML version is definitely the one to use. But for the Browse Happy site, where it is reporting to users about the version of a browser that is available to them, the EdgeHTML version has no meaning.

Likewise, we aren't reporting on the versions of the Gecko or WebKit rendering engines ahead of the versions of their respective browsers (though in the case of the former at least, the browser and engine share major version numbers, unlike Edge and EdgeHTML).

See:
* [Microsoft post on Edge versioning](https://blogs.windows.com/msedgedev/2015/09/21/understanding-versions-in-an-evergreen-browser/)
* [Wikipedia release history for Edge](https://en.wikipedia.org/wiki/Microsoft_Edge#Release_history)

**Comment by mgol** [comment]
@coffee2code If we're talking about regular users only, I wonder if browser versions for rapid release browsers should be included at all. Once a browser goes into such a schema, its versions stop getting advertised and start getting treated as an internal technical detail meant mostly to developers & technical aware people.

Mozilla stopped advertising the Firefox version some time after switching to the rolling release schema (it's not mentioned anywhere at https://www.mozilla.org/firefox/), Opera did it as well once it joined the Chromium cycle; Chrome has always hidden its version, Edge does so as well (IE, which wasn't had a slower cycle, did show its version string to users, urging them to try version 8, 9, 10, in the end 11).

Most people won't even know when to find the version string for non-IE browsers.

**Comment by coredumperror** [comment]
You know, @mgol's answer is really want I should have said. I looked around to find the "25" mentioned in Browser Happy's listing for Edge, but couldn't find it on MS's site. I think browser version numbers have quite honestly become meaningless for end-users. Which means the only people who care about version numbers are developers, but Browser Happy isn't aimed at them.

I agree with mgol: Browser Happy should remove the version numbers from its browser listing for rapid-release browsers. And considering that IE has been discontinued, and Safari is auto-updated by OSX anyway, it's pretty meaningless for those two browsers as well.

**Comment by coffee2code** [comment]
@mgol, @coredumperror: There is definitely merit to the point that browser version numbers have been very much de-emphasized over time. Whether they are worth reporting on the site any longer would be valid to raise in a new issue. Nowadays, I agree, it isn't a very useful piece of information. Happy for either of you to open a new issue for it.

But while we are at the moment reporting versions, I still contend Microsoft Edge's version should trump EdgeHTML's for displaying to the user.

All browsers will report their version number within the browser somewhere (even if not someplace obvious to most users) and thus is technically checkable by them. The EdgeHTML is AFAIK not currently checkable (a user would have to determine their Edge version then cross-reference a resource that shows what EdgeHTML was part of it, or they would have to find a UA parsing site, or something else just as convoluted).

Via the Microsoft post I referenced upthread, they do intend to eventually report the EdgeHTML version number (if they haven't already). Even then, if we reported the EdgeHTML version number instead, it wouldn't be obvious to a user that that version pertains to EdgeHTML (which, by the way, they probably won't know what the heck that is) and not the Edge version.

Anyhow, the debate should be fairly moot since it's looking like we'll just drop reporting browser version numbers altogether.

Thanks for the input and suggestions!
